### PR TITLE
162633505 Ft like or dislike articles

### PR DESCRIPTION
**Pull request by Qadriyah**
#### What does this PR do?
Enables users to like or dislike articles
#### Tasks to be completed
* Like/Unlike articles
* Dislike/Undislike articles
* Write test cases
#### How should this be manually tested
* Run migrations
```
python manage.py migrate
```
* Start server
```
python manage.py runserver
```
* Running tests
```
python manage.py test authors/apps/articles/tests/test_like_articles.py
```
* Endpoint to be tested
```
POST /api/v1/articles/<slug>/like
POST /api/v1/articles/<slug>/dislike
GET /api/v1/articles/
```
* Header
```
Authorization: Get JWT Token from login/register endpoints
```
#### Relevant PT stories
[#162633505](https://www.pivotaltracker.com/n/projects/2231917/stories/162633505)

### CODE REVIEW

**Comment by malep2007** [comment]
@Qadriyah, branch has conflicts, could you first fix those?

**Comment by Qadriyah** [comment]
@malep2007 I have removed the conflicts and updated the PR

File: authors/apps/articles/renderers.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+        errors = data.get('errors', None)
+        custom_errors = {}
+
+        if errors is not None:
+            for key in errors:
+                if key == 'article':
+                    custom_errors.update({
+                        'slug': 'slug is a required field'
+                    })
+                if key == 'like':
```
Comment (jeanjoe) [inline authors/apps/articles/renderers.py]: Add an else part instead to cater for any key outside `article`

**Review by jeanjoe — CHANGES_REQUESTED** [review]

File: authors/apps/articles/serializers.py [hunk 2]
```diff
@@ -1,8 +1,12 @@
-from rest_framework import serializers
+#pylint: disable=E1101
```
Comment (mubstimor) [inline authors/apps/articles/serializers.py]: Is this really necessary?

File: authors/apps/articles/tests/test_articles.py [hunk 3]
```diff
@@ -1,11 +1,13 @@
 from django.test import TestCase
-from  rest_framework.reverse import reverse
+from rest_framework.reverse import reverse
+from rest_framework.reverse import reverse
```
Comment (mubstimor) [inline authors/apps/articles/tests/test_articles.py]: Please remove one of the imports here.

File: authors/apps/articles/renderers.py (hunk 1 above)
Comment (Qadriyah) [reply authors/apps/articles/renderers.py]: I got rid of the renderer. After making the changes, I realised i didn't need it any more

File: authors/apps/articles/serializers.py (hunk 2 above)
Comment (Qadriyah) [reply authors/apps/articles/serializers.py]: Resolved

File: authors/apps/articles/tests/test_articles.py (hunk 3 above)
Comment (Qadriyah) [reply authors/apps/articles/tests/test_articles.py]: Resolved

**Comment by mubstimor** [comment]
LGTM

**Comment by NabunyaLilian** [comment]
LGTM

**Review by jeanjoe — APPROVED** [review]

**Comment by araaliFarooq** [comment]
LGTM

**Comment by mubstimor** [comment]
LGTM

**Comment by mycok** [comment]
LGTM

**Comment by jeanjoe** [comment]
LGTM

File: README.md [hunk 4]
```diff
@@ -15,10 +15,14 @@ by leveraging the modern web.
 if your database does not have a password your can use `postgres://username@localhost:portnumber/database_name`
 - Leave the `SECRET_KEY` as `this-is-hard-to-break`
 
-#### How to configure email
```
Comment (lym) [inline README.md]: Why are you getting rid of these lines?

File: authors/apps/authentication/models.py [hunk 5]
```diff
@@ -141,6 +141,20 @@ def get_short_name(self):
         """
         return self.username
 
+    @property
+    def get_token(self):
```
Comment (lym) [inline authors/apps/authentication/models.py]: Is it really you who worked on this code?

Comment (Qadriyah) [reply authors/apps/authentication/models.py]: Farouk did. I think things got messed up during a rebase operation

File: README.md (hunk 4 above)
Comment (Qadriyah) [reply README.md]: Thank you for the feedback. However this was caused by a rebase operation  but am going to fix it

**Pull request merged**